src/coreComponents/mesh/ParticleManager.hpp

```diff
@@ -0,0 +1,1514 @@
+   * casted to one of the specified subregion types.
+   * @tparam LOOKUP_CONTAINER type of container of names or indices
+   * @tparam LAMBDA type of the user-provided function
+   * @param targetRegions target particle region names or indices
+   * @param lambda kernel function
+   */
+  template< typename SUBREGIONTYPE, typename ... SUBREGIONTYPES, typename LOOKUP_CONTAINER, typename LAMBDA >
+  void forParticleSubRegionsComplete( LOOKUP_CONTAINER const & targetRegions, LAMBDA && lambda )
+  {
+    forParticleRegions( targetRegions, [&] ( localIndex const targetIndex, ParticleRegionBase & particleRegion )
+    {
+      localIndex const er = particleRegion.getIndexInParent();
+
+      if( er>-1 )
+      {
+        for( localIndex esr=0; esr<particleRegion.numSubRegions(); ++esr )
+        {
+          ParticleSubRegionBase & subRegion = particleRegion.getSubRegion( esr );
+
+          Group::applyLambdaToContainer< SUBREGIONTYPE, SUBREGIONTYPES... >( subRegion, [&]( auto & castedSubRegion )
+          {
+            lambda( targetIndex, er, esr, particleRegion, castedSubRegion );
+          } );
+        }
+      }
+    } );
+  }
+
+  /**
+   * @brief This const function is used to launch kernel function over the specified target particle subregions that can
+   * be casted to one of the specified subregion types.
+   * @tparam LOOKUP_CONTAINER type of container of names or indices
+   * @tparam LAMBDA type of the user-provided function
+   * @param targetRegions target particle region names or indices
+   * @param lambda kernel function
+   */
+  template< typename SUBREGIONTYPE, typename ... SUBREGIONTYPES, typename LOOKUP_CONTAINER, typename LAMBDA >
+  void forParticleSubRegionsComplete( LOOKUP_CONTAINER const & targetRegions, LAMBDA && lambda ) const
+  {
+    forParticleRegions( targetRegions, [&] ( localIndex const targetIndex, ParticleRegionBase const & particleRegion )
+    {
+      localIndex const er = particleRegion.getIndexInParent();
+
+      if( er>-1 )
+      {
+        for( localIndex esr=0; esr<particleRegion.numSubRegions(); ++esr )
+        {
+          ParticleSubRegionBase const & subRegion = particleRegion.getSubRegion( esr );
+
+          Group::applyLambdaToContainer< SUBREGIONTYPE, SUBREGIONTYPES... >( subRegion, [&]( auto const & castedSubRegion )
+          {
+            lambda( targetIndex, er, esr, particleRegion, castedSubRegion );
+          } );
+        }
+      }
+    } );
+  }
+
+
+  /**
+   * @brief This is a const function to construct a ParticleViewAccessor to access the data registered on the mesh.
+   * @tparam TRAIT data type
+   * @param neighborName neighbor data name
+   * @return ParticleViewAccessor that contains traits::ViewTypeConst< typename TRAIT::type > data
+   */
+  template< typename TRAIT >
+  ParticleViewAccessor< traits::ViewTypeConst< typename TRAIT::type > >
+  constructExtrinsicAccessor( string const & neighborName = string() ) const;
```

Comment:
It looks like you've built a lot of those helper functions, but I'm not sure you're actually using them.
My IDE does not return some really meaningful results.